mail_layout_force/models/mail_thread.py

```diff
@@ -18,3 +18,52 @@ def message_post_with_template(
         return super().message_post_with_template(
             template_id, email_layout_xmlid=email_layout_xmlid, **kwargs
         )
+
+    def _notify_record_by_email(
+        self,
+        message,
+        recipients_data,
+        msg_vals=False,
+        model_description=False,
+        mail_auto_delete=True,
+        check_existing=False,
+        force_send=True,
+        send_after_commit=True,
+        **kwargs
+    ):
+        msg_vals = msg_vals or {}
+        email_layout_xmlid = (
+            msg_vals.get("email_layout_xmlid")
+            if msg_vals
+            else message.email_layout_xmlid
+        )
+        template_xmlid = (
+            email_layout_xmlid
+            if email_layout_xmlid
+            else "mail.message_notification_email"
+        )
+        template = self.env.ref(template_xmlid, raise_if_not_found=True)
+        res_model = (
+            self.env["ir.model"].sudo().search([("model", "=", self._name)], limit=1)
+        )
+        layout_mapping = self.env["email.layout.mapping"].search(
+            [("layout_id", "=", template.id), ("model_ids", "in", res_model.ids)],
+            limit=1,
+        )
+        if not layout_mapping:
+            layout_mapping = self.env["email.layout.mapping"].search(
+                [("layout_id", "=", template.id), ("model_ids", "=", False)], limit=1
+            )
+        if layout_mapping:
+            msg_vals["email_layout_xmlid"] = layout_mapping.substitute_layout_id.xml_id
```

Comment:
Can we switch to use _export_rows().
```suggestion
        layout_xmlid = (
            msg_vals.get("email_layout_xmlid")
            or message.email_layout_xmlid
            or "mail.message_notification_email"
        )
        layout = self.env.ref(layout_xmlid, raise_if_not_found=True)
        res_model = (
            self.env["ir.model"].sudo().search([("model", "=", self._name)], limit=1)
        )
        mapping = self.env["email.layout.mapping"].search(
            [("layout_id", "=", layout.id), ("model_ids", "in", res_model.ids)],
            limit=1,
        )
        if not mapping:
            mapping = self.env["email.layout.mapping"].search(
                [("layout_id", "=", layout.id), ("model_ids", "=", False)], limit=1
            )
        if mapping:
            substitute_layout = mapping.substitute_layout_id
            if not substitute_layout.xml_id:
                substitute_layout._export_rows([["id"]])
            msg_vals["email_layout_xmlid"] = mapping.substitute_layout_id.xml_id
```